### PR TITLE
Allow exposing domains in cloud

### PR DESCRIPTION
**Pull request by bramkragten**
https://github.com/home-assistant/core/pull/39216

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/cloud/alexa/cloud-alexa.ts [hunk 1]
```diff
@@ -117,33 +125,76 @@ class CloudAlexa extends LitElement {
...
+                      ? isExposed
+                        ? mdiCheckboxMarked
+                        : mdiCloseBox
+                      : isDomainExposed
+                      ? mdiCheckboxMultipleMarked
+                      : mdiCloseBoxMultiple}
+                  ></ha-svg-icon>
+                </mwc-icon-button>
+                <mwc-list-item hasMeta>
+                  Expose entity
```
Comment (ludeeus) [inline src/panels/config/cloud/alexa/cloud-alexa.ts:168]:
```suggestion
                  ${this.hass!.localize("ui.panel.config.cloud.alexa.expose_entity")}
```

File: src/panels/config/cloud/alexa/cloud-alexa.ts [hunk 2]
```diff
@@ -117,33 +125,76 @@ class CloudAlexa extends LitElement {
...
+                  <ha-svg-icon
+                    class="exposed"
+                    slot="meta"
+                    .path=${mdiCheckboxMarked}
+                  ></ha-svg-icon>
+                </mwc-list-item>
+                <mwc-list-item hasMeta>
+                  Don't expose entity
```
Comment (ludeeus) [inline src/panels/config/cloud/alexa/cloud-alexa.ts:176]:
```suggestion
                  ${this.hass!.localize("ui.panel.config.cloud.alexa.dont_expose_entity")}
```

File: src/panels/config/cloud/alexa/cloud-alexa.ts [hunk 3]
```diff
@@ -117,33 +125,76 @@ class CloudAlexa extends LitElement {
...
+                  <ha-svg-icon
+                    class="not-exposed"
+                    slot="meta"
+                    .path=${mdiCloseBox}
+                  ></ha-svg-icon>
+                </mwc-list-item>
+                <mwc-list-item hasMeta>
+                  Follow domain
```
Comment (ludeeus) [inline src/panels/config/cloud/alexa/cloud-alexa.ts:184]:
```suggestion
                  ${this.hass!.localize("ui.panel.config.cloud.alexa.follow_domain")}
```

File: src/panels/config/cloud/alexa/cloud-alexa.ts [hunk 4]
```diff
@@ -157,17 +208,14 @@ class CloudAlexa extends LitElement {
...
         ${
           emptyFilter
             ? html`
-                <ha-icon-button
+                <mwc-button
                   slot="toolbar-icon"
-                  icon="hass:tune"
                   @click=${this._openDomainToggler}
-                ></ha-icon-button>
+                  >Manage domains</mwc-button
```
Comment (ludeeus) [inline src/panels/config/cloud/alexa/cloud-alexa.ts:217]:
```suggestion
                  >${this.hass!.localize("ui.panel.config.cloud.alexa.manage_domains")}</mwc-button
```

File: src/panels/config/cloud/alexa/cloud-alexa.ts [hunk 5]
```diff
@@ -183,23 +231,31 @@ class CloudAlexa extends LitElement {
...
-                      "ui.panel.config.cloud.alexa.exposed_entities"
-                    )}
-                  </h1>
+                  <div class="header">
+                    <h3>
+                      ${this.hass!.localize(
+                        "ui.panel.config.cloud.alexa.exposed_entities"
+                      )}
+                    </h3>
+                    ${selected}${!this.narrow ? html` exposed` : ""}
```
Comment (ludeeus) [inline src/panels/config/cloud/alexa/cloud-alexa.ts:240]:
```suggestion
                    ${selected}${!this.narrow ? html` ${this.hass!.localize("ui.panel.config.cloud.alexa.exposed")}` : ""}
```

File: src/panels/config/cloud/alexa/cloud-alexa.ts [hunk 6]
```diff
@@ -183,23 +231,31 @@ class CloudAlexa extends LitElement {
...
-                    )}
-                  </h1>
+                  <div class="header second">
+                    <h3>
+                      ${this.hass!.localize(
+                        "ui.panel.config.cloud.alexa.not_exposed_entities"
+                      )}
+                    </h3>
+                    ${this._entities.length - selected}${!this.narrow
+                      ? html` not exposed`
```
Comment (ludeeus) [inline src/panels/config/cloud/alexa/cloud-alexa.ts:256]:
```suggestion
                      ? html` ${this.hass!.localize("ui.panel.config.cloud.alexa.not_exposed")}`
```

File: src/panels/config/cloud/google-assistant/cloud-google-assistant.ts [hunk 7]
```diff
@@ -121,31 +131,72 @@ class CloudGoogleAssistant extends LitElement {
...
+                      ? isExposed
+                        ? mdiCheckboxMarked
+                        : mdiCloseBox
+                      : isDomainExposed
+                      ? mdiCheckboxMultipleMarked
+                      : mdiCloseBoxMultiple}
+                  ></ha-svg-icon>
+                </mwc-icon-button>
+                <mwc-list-item hasMeta>
+                  Expose entity
```
Comment (ludeeus) [inline src/panels/config/cloud/google-assistant/cloud-google-assistant.ts:171]: localize

File: src/panels/config/cloud/google-assistant/cloud-google-assistant.ts [hunk 8]
```diff
@@ -179,16 +230,16 @@ class CloudGoogleAssistant extends LitElement {
         "ui.panel.config.cloud.google.title"
       )}">
         <span slot="toolbar-icon">
-          ${selected}${!this.narrow ? html` selected ` : ""}
+          
         </span>
```
Comment (ludeeus) [inline src/panels/config/cloud/google-assistant/cloud-google-assistant.ts:234]: This is now empty, can it be removed?

**Review by ludeeus — CHANGES_REQUESTED** [review]
Generally looks good :+1:, but we should localize where we can.
Also added a few other sugestions.

File: src/panels/config/cloud/alexa/cloud-alexa.ts [hunk 9]
```diff
@@ -117,33 +125,76 @@ class CloudAlexa extends LitElement {
...
+                  class=${classMap({
+                    exposed: isExposed!,
+                    "not-exposed": !isExposed,
+                  })}
+                  .title=${this.hass!.localize(
+                    "ui.panel.config.cloud.alexa.expose"
+                  )}
+                >
+                  <ha-svg-icon
+                    path=${config.should_expose !== null
```
Comment (zsarnett) [inline src/panels/config/cloud/alexa/cloud-alexa.ts:158]:
```suggestion
                    .path=${config.should_expose !== null
```

File: src/panels/config/cloud/alexa/cloud-alexa.ts [hunk 10]
```diff
@@ -183,23 +231,31 @@ class CloudAlexa extends LitElement {
...
-                      "ui.panel.config.cloud.alexa.exposed_entities"
-                    )}
-                  </h1>
+                  <div class="header">
+                    <h3>
+                      ${this.hass!.localize(
+                        "ui.panel.config.cloud.alexa.exposed_entities"
+                      )}
+                    </h3>
+                    ${selected}${!this.narrow ? html` ${this.hass!.localize("ui.panel.config.cloud.alexa.exposed")}` : ""}
```
Comment (zsarnett) [inline src/panels/config/cloud/alexa/cloud-alexa.ts:240]:
```suggestion
                    ${selected}${!this.narrow ? html`${this.hass!.localize("ui.panel.config.cloud.alexa.exposed")}` : ""}
```

File: src/panels/config/cloud/alexa/cloud-alexa.ts [hunk 11]
```diff
@@ -183,23 +231,31 @@ class CloudAlexa extends LitElement {
...
-                    )}
-                  </h1>
+                  <div class="header second">
+                    <h3>
+                      ${this.hass!.localize(
+                        "ui.panel.config.cloud.alexa.not_exposed_entities"
+                      )}
+                    </h3>
+                    ${this._entities.length - selected}${!this.narrow
+                      ? html` ${this.hass!.localize("ui.panel.config.cloud.alexa.not_exposed")}`
```
Comment (zsarnett) [inline src/panels/config/cloud/alexa/cloud-alexa.ts:256]:
```suggestion
                      ? html`${this.hass!.localize("ui.panel.config.cloud.alexa.not_exposed")}`
```

File: src/panels/config/cloud/google-assistant/cloud-google-assistant.ts [hunk 12]
```diff
@@ -121,31 +131,72 @@ class CloudGoogleAssistant extends LitElement {
...
+                    "not-exposed": !isExposed,
+                  })}
+                  .title=${this.hass!.localize(
+                    "ui.panel.config.cloud.google.expose"
+                  )}
                 >
-                </ha-switch>
-              </ha-formfield>
+                  <ha-svg-icon
+                    path=${config.should_expose !== null
```
Comment (zsarnett) [inline src/panels/config/cloud/google-assistant/cloud-google-assistant.ts:161]:
```suggestion
                    .path=${config.should_expose !== null
```

File: src/dialogs/domain-toggler/dialog-domain-toggler.ts [hunk 13]
```diff
@@ -35,55 +42,49 @@ class DomainTogglerDialog extends LitElement {
...
 
     return html`
-      <ha-paper-dialog
-        with-backdrop
-        opened
-        @opened-changed=${this._openedChanged}
+      <ha-dialog
+        open
+        @closed=${this.closeDialog}
+        .heading=${this.hass.localize("ui.dialogs.domain_toggler.title")}
```
Comment (ludeeus) [inline src/dialogs/domain-toggler/dialog-domain-toggler.ts:48]:
```suggestion
        scrimClickAction
        escapeKeyAction
        hideActions
        .heading=${createCloseHeading(
          this.hass,
          this.hass.localize("ui.dialogs.domain_toggler.title")
        )}
```

File: src/dialogs/domain-toggler/dialog-domain-toggler.ts [hunk 14]
```diff
@@ -4,27 +4,34 @@ import {
...
   customElement,
   html,
-  LitElement,
   internalProperty,
+  LitElement,
   TemplateResult,
 } from "lit-element";
-import "../../components/dialog/ha-paper-dialog";
+import { fireEvent } from "../../common/dom/fire_event";
+import "../../components/ha-dialog";
```
Comment (ludeeus) [inline src/dialogs/domain-toggler/dialog-domain-toggler.ts:12]:
```suggestion
import { createCloseHeading } from "../../components/ha-dialog";
```

File: src/panels/config/cloud/alexa/cloud-alexa.ts [hunk 15]
```diff
@@ -117,33 +125,80 @@ class CloudAlexa extends LitElement {
...
-              </ha-switch>
-            </ha-formfield>
+                <mwc-icon-button
+                  slot="trigger"
+                  class=${classMap({
+                    exposed: isExposed!,
+                    "not-exposed": !isExposed,
+                  })}
+                  .title=${this.hass!.localize(
+                    "ui.panel.config.cloud.alexa.expose"
```
Comment (ludeeus) [inline src/panels/config/cloud/alexa/cloud-alexa.ts:152]: I know I said to call it this, but now we have identical keys for google/alexa, should we combine them?

File: src/dialogs/domain-toggler/dialog-domain-toggler.ts [hunk 16]
```diff
@@ -35,55 +42,49 @@ class DomainTogglerDialog extends LitElement {
...
+        .heading=${this.hass.localize("ui.dialogs.domain_toggler.title")}
       >
-        <h2>
-          ${this.hass.localize("ui.dialogs.domain_toggler.title")}
-        </h2>
         <div>
           ${domains.map(
             (domain) =>
               html`
                 <div>${domain[0]}</div>
```
Comment (ludeeus) [inline src/dialogs/domain-toggler/dialog-domain-toggler.ts:54]:
```suggestion
                <ha-formfield .label=${domain[0]}>
```

File: src/dialogs/domain-toggler/dialog-domain-toggler.ts [hunk 17]
```diff
@@ -35,55 +42,49 @@ class DomainTogglerDialog extends LitElement {
...
-                </mwc-button>
-                <mwc-button .domain=${domain[1]} @click=${this._handleOn}>
-                  ${this.hass.localize("state.default.on")}
+                <ha-switch
+                  .domain=${domain[1]}
+                  .checked=${!this._params!.exposedDomains ||
+                  this._params!.exposedDomains.includes(domain[1])}
+                  @change=${this._handleSwitch}
+                >
+                </ha-switch>
```
Comment (ludeeus) [inline src/dialogs/domain-toggler/dialog-domain-toggler.ts:61]:
```suggestion
                </ha-switch>
              </ha-formfield>
```

File: src/dialogs/domain-toggler/dialog-domain-toggler.ts [hunk 18]
```diff
@@ -4,27 +4,34 @@ import {
...
   html,
-  LitElement,
   internalProperty,
+  LitElement,
   TemplateResult,
 } from "lit-element";
-import "../../components/dialog/ha-paper-dialog";
+import { fireEvent } from "../../common/dom/fire_event";
+import "../../components/ha-dialog";
+import "../../components/ha-switch";
```
Comment (ludeeus) [inline src/dialogs/domain-toggler/dialog-domain-toggler.ts:13]:
```suggestion
import "../../components/ha-switch";
import "../../components/ha-formfield";
```

File: src/panels/config/cloud/alexa/cloud-alexa.ts (hunk 15 above)
Comment (bramkragten) [reply src/panels/config/cloud/alexa/cloud-alexa.ts:152]: I want to combine the two pages in the future, as they are almost identical

**Review by ludeeus — APPROVED** [review]
Nice! :tada:

**Pull request merged**